flang/include/flang/Optimizer/Builder/Runtime/Command.h

```diff
@@ -16,12 +16,27 @@ class Location;
 
 namespace fir {
 class FirOpBuilder;
+class CharBoxValue;
+} // namespace fir
+
+namespace llvm {
+template <typename T>
+class Optional;
 }
 
 namespace fir::runtime {
 
 /// Generate call to COMMAND_ARGUMENT_COUNT intrinsic runtime routine.
 mlir::Value genCommandArgumentCount(fir::FirOpBuilder &, mlir::Location);
 
+/// Generate call to GET_COMMAND_ARGUMENT intrinsic runtime routine.
+/// Note that GET_COMMAND_ARGUMENT intrinsic is split between 2 functions in
+/// implementation; ArgumentValue and ArgumentLength. So we handle each
+/// seperately.
```

Comment:
[nit] To me this is an implementation detail
>  Note that GET_COMMAND_ARGUMENT intrinsic is split between 2 functions in implementation; ArgumentValue and ArgumentLength. So we handle each separately.

In a header file like this one I'd normally focus on the interface. The implementation might change in the future, but since its in a different file, people  might forget to update this comment.